### PR TITLE
Home Work 4

### PR DESCRIPTION
**Pull request by mykola-mizin**

### CODE REVIEW

File: build_in_methods_iterators/homework.py [hunk 1]
```diff
@@ -8,24 +12,30 @@
...
-        fix_names_start_letters([{'name': 'Alex', 'age': 26}, {'name': 'denys', 'age': 89}])
-        >>> [{'name': 'Alex', 'age': 26}, {'name': 'Denys', 'age': 89}]
+       Make all `names` field in list of students to start from upper letter
+       Examples:
+           fix_names_start_letters([{'name': 'Alex', 'age': 26}, {'name': 'denys', 'age': 89}])
+           >>> [{'name': 'Alex', 'age': 26}, {'name': 'Denys', 'age': 89}]
     """
-    pass
+
+    [k.update({'name': k.get('name').capitalize()}) for k in data if k.get('name')]
```
Comment (dimigor) [inline build_in_methods_iterators/homework.py]: You should use simple loop

File: build_in_methods_iterators/homework.py [hunk 2]
```diff
@@ -8,24 +12,30 @@
...
     """given_data
     Remove from dictionaries given key value
 
     Examples:
        remove_dict_field([{'name': 'Alex', 'age': 26}, {'name': 'denys', 'age': 89}], 'age')
         >>> [{'name': 'Alex'}, {'name': 'denys'}]
     """
-    pass
+    for i in range(len(redundant_keys)):
+        [k.pop(redundant_keys[i])  for k in data]
```
Comment (dimigor) [inline build_in_methods_iterators/homework.py]: the same as above

File: build_in_methods_iterators/homework.py [hunk 3]
```diff
@@ -58,21 +75,22 @@ def task_6_min_value_list_of_dicts(data: DT, key: str) -> ST:
     Returns:
 
     """
-    pass
+    return dict(*[i for i in data if i.get(key)==(min([i.get(key) for i in data if i.get(key)]))])
```
Comment (dimigor) [inline build_in_methods_iterators/homework.py]: this line contains too much logic
rather write more lines of code and readable of your code increase

File: build_in_methods_iterators/homework.py (hunk 1 above)
Comment (mykola-mizin) [reply build_in_methods_iterators/homework.py]: done

File: build_in_methods_iterators/homework.py (hunk 2 above)
Comment (mykola-mizin) [reply build_in_methods_iterators/homework.py]: done

File: build_in_methods_iterators/homework.py (hunk 3 above)
Comment (mykola-mizin) [reply build_in_methods_iterators/homework.py]: done

File: build_in_methods_iterators/tests.py [hunk 4]
```diff
@@ -198,4 +198,4 @@ def test_task_3_one_result(self):
 
 
 if __name__ == "__main__":
-    unittest.main()
+    unittest.main()
```
Comment (dimigor) [inline build_in_methods_iterators/tests.py]: add empty line please)

Comment (mykola-mizin) [reply build_in_methods_iterators/tests.py]: done

**Pull request merged**